src/app/pages/AccountDetailsPage/hook.ts

```diff
@@ -0,0 +1,51 @@
+import { Layer, useGetRuntimeAccountsAddress, useGetRuntimeTransactions } from '../../../oasis-indexer/api'
+import { AppErrors } from '../../../types/errors'
+import { useSearchParamsPagination } from '../../components/Table/useSearchParamsPagination'
+import { NUMBER_OF_ITEMS_ON_SEPARATE_PAGE } from '../../config'
+import { SearchScope } from '../../../types/searchScope'
+
+export const useAccount = (scope: SearchScope, address: string) => {
+  const { network, layer } = scope
+  if (layer === Layer.consensus) {
+    // There can be no ERC-20 or ERC-721 tokens on consensus
+    throw AppErrors.UnsupportedLayer
+  }
+  const query = useGetRuntimeAccountsAddress(network, layer, address!)
+  const account = query.data?.data
+  const isLoading = query.isLoading
+  return { account, isLoading }
+}
+
+export const useTransactions = (scope: SearchScope, address: string) => {
+  const { network, layer } = scope
+  const pagination = useSearchParamsPagination('page')
+  const offset = (pagination.selectedPage - 1) * NUMBER_OF_ITEMS_ON_SEPARATE_PAGE
+  if (layer === Layer.consensus) {
+    throw AppErrors.UnsupportedLayer
+    // Loading transactions on the consensus layer is not supported yet.
+    // We should use useGetConsensusTransactions()
+  }
+  const query = useGetRuntimeTransactions(
+    network,
+    layer, // This is OK since consensus has been handled separately
+    {
+      limit: NUMBER_OF_ITEMS_ON_SEPARATE_PAGE,
+      offset: offset,
+      rel: address,
+    },
+  )
+  const { isFetched, isLoading, data } = query
+
+  const transactions = data?.data.transactions || []
```

Comment:
This broke loading state, and I think it's usually good to avoid these fallbacks

Was there a reason for this one?

Reply:
Why is loading state not detected using the dedicated flag? In any case I will check it

Reply:
To answer my question: I gather that Table wants to keep the existing data even while loading, and only show the skeleton rows when we are loading, _and_ we have no previous data. This is why setting `isLoading` to true is not sufficient to trigger the skeleton rows.

Reply:
Fix is available at https://github.com/oasisprotocol/explorer/pull/475